decode.go

```diff
@@ -102,7 +102,7 @@ func (p *parser) peek() yamlEventType {
 	// It's curious choice from the underlying API to generally return a
 	// positive result on success, but on this case return true in an error
 	// scenario. This was the source of bugs in the past (issue #666).
-	if !yaml_parser_parse(&p.parser, &p.event) || p.parser.error != yaml_NO_ERROR {
+	if !(&p.parser).parse(&p.event) || p.parser.error != yaml_NO_ERROR {
```

Comment:
This is not idiomatic

Here I feel like parse should return an error

Reply:
probably OOS for this PR?

Reply:
Indeed